magenta/models/nsynth/wavenet/fastgen.py

```diff
@@ -171,14 +177,27 @@ def synthesize(encodings,
     checkpoint_path: Location of the pretrained model. [model.ckpt-200000]
     samples_per_save: Save files after every amount of generated samples.
   """
+  trace_filename = FLAGS.trace_file
+  if trace_filename: 
+    run_options = tf.RunOptions(trace_level=tf.RunOptions.FULL_TRACE)
+    run_metadata = tf.RunMetadata()
+  else:
+    run_options = None 
+    run_metadata = None 
+
   hop_length = Config().ae_hop_length
   # Get lengths
   batch_size = encodings.shape[0]
   encoding_length = encodings.shape[1]
   total_length = encoding_length * hop_length
 
   session_config = tf.ConfigProto(allow_soft_placement=True)
+  #inter and intra parallelism for CPU
```

Comment:
Can you make all of this configuration optional depending on whether the flags are specified?